src/App.tsx

```diff
@@ -1,61 +1,146 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+const addUserTodo: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodo] = useState(addUserTodo);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [validInputTitle, setValidInputTitle] = useState(false);
```

Comment:
```suggestion
  const [isValidTitle, setIsValidTitle] = useState(false);
```
Name boolean variables according to these rules:
![image](https://user-images.githubusercontent.com/96409563/224632371-cdd7515b-3872-4eb1-87bc-ce463922403d.png)
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig